### ISSUE TITLE
The v23.4.0 VC constantly disconnects and reconnects from a Prysm BN

### ISSUE THREAD

**Issue by jclapis** [status-im/nim-beacon-chain#4872]
**Describe the bug**
Hey guys, I just upgraded to v23.4.0 and have the following configuration:

- Local Nimbus BN (v23.4.0)
- Local Nimbus VC (v23.4.0)
- Remote Prysm BN (v4.0.3)

The VC is attached to both BNs. Starting with v23.4.0, the VC constantly disconnects from the Prysm BN and reconnects, producing the following logs:

```
rocketpool_validator  | WRN 2023-04-28 08:26:04.002+00:00 Beacon node down                           node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | INF 2023-04-28 08:26:12.001+00:00 Slot start                                 slot=5513531 epoch=172297 attestationIn=4s blockIn=<unknown> validators=46 good_nodes=1 viable_nodes=0 bad_nodes=1 delay=1ms217us575ns
rocketpool_validator  | NOT 2023-04-28 08:26:13.121+00:00 Beacon node is online                      agent_version="prysm/v4.0.3 (linux arm64)" node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | NOT 2023-04-28 08:26:13.130+00:00 Beacon node is compatible                  node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | NOT 2023-04-28 08:26:13.132+00:00 Beacon node is in sync                     head_slot=5513530 sync_distance=1 is_optimistic=false node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | WRN 2023-04-28 08:26:16.003+00:00 Beacon node down                           node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | NOT 2023-04-28 08:26:16.160+00:00 Attestation published                      delay=161ms34us888ns service=attestation_service validator=8d696c7c@445010 attestation="(aggregation_bits: 0b00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, data: (slot: 5513531, index: 45, beacon_block_root: \"2fcb0bc0\", source: \"172296:16e85d8a\", target: \"172297:73cb6dcf\"), signature: \"b0beb24b\")"
rocketpool_validator  | NOT 2023-04-28 08:26:19.142+00:00 Beacon node is online                      agent_version="prysm/v4.0.3 (linux arm64)" node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | NOT 2023-04-28 08:26:19.148+00:00 Beacon node is compatible                  node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | NOT 2023-04-28 08:26:19.151+00:00 Beacon node is in sync                     head_slot=5513530 sync_distance=1 is_optimistic=false node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | INF 2023-04-28 08:26:24.121+00:00 Slot start                                 slot=5513532 epoch=172297 attestationIn=16s blockIn=<unknown> validators=46 good_nodes=2 viable_nodes=0 bad_nodes=0 delay=121ms52us498ns
rocketpool_validator  | WRN 2023-04-28 08:26:28.002+00:00 Beacon node down                           node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | INF 2023-04-28 08:26:36.001+00:00 Slot start                                 slot=5513533 epoch=172297 attestationIn=4s blockIn=<unknown> validators=46 good_nodes=1 viable_nodes=0 bad_nodes=1 delay=1ms333us619ns
rocketpool_validator  | NOT 2023-04-28 08:26:37.177+00:00 Beacon node is online                      agent_version="prysm/v4.0.3 (linux arm64)" node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | NOT 2023-04-28 08:26:37.187+00:00 Beacon node is compatible                  node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | NOT 2023-04-28 08:26:37.885+00:00 Beacon node is in sync                     head_slot=5513533 sync_distance=0 is_optimistic=false node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
``` 

This is reproducible on both Goerli and Mainnet.

Running the VC with the debug logs on reveals the following error while sending REST headers apparently?

```
rocketpool_validator  | NOT 2023-04-28 08:29:01.635+00:00 Beacon node is in sync                     head_slot=5513545 sync_distance=0 is_optimistic=false node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:04.000+00:00 Producing sync committee messages          delay=512us302ns slot=5513545 duties_count=0 service=sync_committee_service
rocketpool_validator  | DBG 2023-04-28 08:29:04.002+00:00 Sending REST request to remote server      remote=eth2:5052 request=/eth/v1/beacon/blocks/head/root http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:04.003+00:00 Sending REST request to remote server      remote=192.168.1.187:5052 request=/eth/v1/beacon/blocks/head/root http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:04.005+00:00 REST client communication error            remote=192.168.1.187:5052 request=/eth/v1/beacon/blocks/head/root connection=9 errorName=HttpWriteError errorMsg="Could not send request headers"
rocketpool_validator  | DBG 2023-04-28 08:29:04.007+00:00 Unable to retrieve head block's root       node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD error="[RestCommunicationError] Communication failed while sending/receiving request, http error [HttpWriteError]: Could not send request headers" strategy=first request=getHeadBlockRoot
rocketpool_validator  | WRN 2023-04-28 08:29:04.008+00:00 Beacon node down                           node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:04.009+00:00 Got REST response headers from remote server status=200 http_method=GET remote=eth2:5052 request=/eth/v1/beacon/blocks/head/root connection=1
rocketpool_validator  | DBG 2023-04-28 08:29:04.011+00:00 Received REST response body from remote server remote=eth2:5052 request=/eth/v1/beacon/blocks/head/root contentType=application/json size=115 connection=1
rocketpool_validator  | DBG 2023-04-28 08:29:04.012+00:00 Sync committee message statistics          total=0 succeed=0 failed_to_deliver=0 not_accepted=0 delay=12ms359us981ns slot=5513545 duties_count=0 service=sync_committee_service
rocketpool_validator  | DBG 2023-04-28 08:29:05.638+00:00 Checking beacon node                       node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD status=offline service=fallback_service
rocketpool_validator  | DBG 2023-04-28 08:29:05.639+00:00 Checking beacon node status                service=fallback_service node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:05.639+00:00 Sending REST request to remote server      remote=192.168.1.187:5052 request=/eth/v1/node/version http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:05.639+00:00 REST client communication error            remote=192.168.1.187:5052 request=/eth/v1/node/version connection=10 errorName=HttpWriteError errorMsg="Could not send request headers"
rocketpool_validator  | DBG 2023-04-28 08:29:05.640+00:00 Unable to check beacon node's status       error_name=RestCommunicationError error_message="Communication failed while sending/receiving request, http error [HttpWriteError]: Could not send request headers" service=fallback_service node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:07.643+00:00 Checking beacon node                       node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD status=offline service=fallback_service
rocketpool_validator  | DBG 2023-04-28 08:29:07.643+00:00 Checking beacon node status                service=fallback_service node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:07.643+00:00 Sending REST request to remote server      remote=192.168.1.187:5052 request=/eth/v1/node/version http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:07.643+00:00 REST client communication error            remote=192.168.1.187:5052 request=/eth/v1/node/version connection=11 errorName=HttpWriteError errorMsg="Could not send request headers"
rocketpool_validator  | DBG 2023-04-28 08:29:07.644+00:00 Unable to check beacon node's status       error_name=RestCommunicationError error_message="Communication failed while sending/receiving request, http error [HttpWriteError]: Could not send request headers" service=fallback_service node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:08.001+00:00 Producing contribution and proofs          delay=1ms188us457ns service=sync_committee_service
rocketpool_validator  | DBG 2023-04-28 08:29:08.001+00:00 No contribution and proofs scheduled for slot slot=5513545 service=sync_committee_service
rocketpool_validator  | DBG 2023-04-28 08:29:09.645+00:00 Checking beacon node                       node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD status=offline service=fallback_service
rocketpool_validator  | DBG 2023-04-28 08:29:09.645+00:00 Checking beacon node status                service=fallback_service node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:09.645+00:00 Sending REST request to remote server      remote=192.168.1.187:5052 request=/eth/v1/node/version http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:09.646+00:00 REST client communication error            remote=192.168.1.187:5052 request=/eth/v1/node/version connection=12 errorName=HttpWriteError errorMsg="Could not send request headers"
rocketpool_validator  | DBG 2023-04-28 08:29:09.647+00:00 Unable to check beacon node's status       error_name=RestCommunicationError error_message="Communication failed while sending/receiving request, http error [HttpWriteError]: Could not send request headers" service=fallback_service node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:11.648+00:00 Checking beacon node                       node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD status=offline service=fallback_service
rocketpool_validator  | DBG 2023-04-28 08:29:11.649+00:00 Checking beacon node status                service=fallback_service node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:11.649+00:00 Sending REST request to remote server      remote=192.168.1.187:5052 request=/eth/v1/node/version http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:11.649+00:00 REST client communication error            remote=192.168.1.187:5052 request=/eth/v1/node/version connection=13 errorName=HttpWriteError errorMsg="Could not send request headers"
rocketpool_validator  | DBG 2023-04-28 08:29:11.650+00:00 Unable to check beacon node's status       error_name=RestCommunicationError error_message="Communication failed while sending/receiving request, http error [HttpWriteError]: Could not send request headers" service=fallback_service node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | INF 2023-04-28 08:29:12.005+00:00 Slot start                                 slot=5513546 epoch=172298 attestationIn=16s blockIn=<unknown> validators=46 good_nodes=1 viable_nodes=0 bad_nodes=1 delay=5ms494us54ns
rocketpool_validator  | DBG 2023-04-28 08:29:12.005+00:00 Sending REST request to remote server      remote=eth2:5052 request=/eth/v1/validator/duties/proposer/172298 http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:12.006+00:00 Sending REST request to remote server      remote=eth2:5052 request=/eth/v1/validator/duties/attester/172298 http_method=POST
rocketpool_validator  | DBG 2023-04-28 08:29:12.007+00:00 Opened connection to remote server         remote=eth2:5052 request=/eth/v1/validator/duties/attester/172298 http_method=POST connection=2
rocketpool_validator  | DBG 2023-04-28 08:29:12.008+00:00 Sending REST request to remote server      remote=eth2:5052 request="/eth/v1/beacon/states/head/validators?id=0xa0469d1ef94130809f30a5f953fba6b511771e7c575bb20d49a7526125a0fbc0408420d29977310c8f85606366589b0b" http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:12.102+00:00 Validator registration missing validator index validator=a0469d1e
rocketpool_validator  | DBG 2023-04-28 08:29:12.154+00:00 Validator registrations prepared           total=46 succeed=45 cached=0 bad=0 errors=0 index_missing=1 fee_missing=0 incorrect_time=0
rocketpool_validator  | DBG 2023-04-28 08:29:12.155+00:00 Sending REST request to remote server      remote=eth2:5052 request=/eth/v1/validator/register_validator http_method=POST
rocketpool_validator  | DBG 2023-04-28 08:29:12.156+00:00 Opened connection to remote server         remote=eth2:5052 request=/eth/v1/validator/register_validator http_method=POST connection=4
rocketpool_validator  | DBG 2023-04-28 08:29:12.160+00:00 Sending REST request to remote server      remote=eth2:5052 request=/eth/v1/validator/prepare_beacon_proposer http_method=POST
rocketpool_validator  | DBG 2023-04-28 08:29:12.160+00:00 Opened connection to remote server         remote=eth2:5052 request=/eth/v1/validator/prepare_beacon_proposer http_method=POST connection=5
rocketpool_validator  | DBG 2023-04-28 08:29:12.160+00:00 Sending REST request to remote server      remote=eth2:5052 request=/eth/v1/validator/duties/sync/172298 http_method=POST
rocketpool_validator  | DBG 2023-04-28 08:29:12.161+00:00 Opened connection to remote server         remote=eth2:5052 request=/eth/v1/validator/duties/sync/172298 http_method=POST connection=6
rocketpool_validator  | DBG 2023-04-28 08:29:12.162+00:00 REST request body has been sent            remote=eth2:5052 request=/eth/v1/validator/duties/attester/172298 size=406 http_method=POST connection=2
rocketpool_validator  | DBG 2023-04-28 08:29:12.162+00:00 REST request body has been sent            remote=eth2:5052 request=/eth/v1/validator/prepare_beacon_proposer size=4051 http_method=POST connection=5
rocketpool_validator  | DBG 2023-04-28 08:29:12.162+00:00 REST request body has been sent            remote=eth2:5052 request=/eth/v1/validator/duties/sync/172298 size=406 http_method=POST connection=6
rocketpool_validator  | DBG 2023-04-28 08:29:12.163+00:00 Got REST response headers from remote server status=200 http_method=GET remote=eth2:5052 request=/eth/v1/validator/duties/proposer/172298 connection=1
rocketpool_validator  | DBG 2023-04-28 08:29:12.163+00:00 Got REST response headers from remote server status=200 http_method=GET remote=eth2:5052 request="/eth/v1/beacon/states/head/validators?id=0xa0469d1ef94130809f30a5f953fba6b511771e7c575bb20d49a7526125a0fbc0408420d29977310c8f85606366589b0b" connection=3
rocketpool_validator  | DBG 2023-04-28 08:29:12.164+00:00 Received REST response body from remote server remote=eth2:5052 request="/eth/v1/beacon/states/head/validators?id=0xa0469d1ef94130809f30a5f953fba6b511771e7c575bb20d49a7526125a0fbc0408420d29977310c8f85606366589b0b" contentType=application/json size=40 connection=3
rocketpool_validator  | DBG 2023-04-28 08:29:12.165+00:00 REST request body has been sent            remote=eth2:5052 request=/eth/v1/validator/register_validator size=19891 http_method=POST connection=4
rocketpool_validator  | DBG 2023-04-28 08:29:12.166+00:00 Received REST response body from remote server remote=eth2:5052 request=/eth/v1/validator/duties/proposer/172298 contentType=application/json size=5111 connection=1
rocketpool_validator  | DBG 2023-04-28 08:29:12.166+00:00 No relevant proposer duties received       slot=5513546 duties_count=32 service=duties_service
rocketpool_validator  | DBG 2023-04-28 08:29:12.179+00:00 Got REST response headers from remote server status=200 http_method=POST remote=eth2:5052 request=/eth/v1/validator/duties/attester/172298 connection=2
rocketpool_validator  | DBG 2023-04-28 08:29:12.181+00:00 Got REST response headers from remote server status=200 http_method=POST remote=eth2:5052 request=/eth/v1/validator/prepare_beacon_proposer connection=5
rocketpool_validator  | DBG 2023-04-28 08:29:12.183+00:00 Got REST response headers from remote server status=200 http_method=POST remote=eth2:5052 request=/eth/v1/validator/duties/sync/172298 connection=6
rocketpool_validator  | DBG 2023-04-28 08:29:12.194+00:00 Received REST response body from remote server remote=eth2:5052 request=/eth/v1/validator/prepare_beacon_proposer contentType=text/plain size=0 connection=5
rocketpool_validator  | DBG 2023-04-28 08:29:12.196+00:00 Received REST response body from remote server remote=eth2:5052 request=/eth/v1/validator/duties/attester/172298 contentType=application/json size=4342 connection=2
rocketpool_validator  | DBG 2023-04-28 08:29:12.198+00:00 Sending REST request to remote server      remote=eth2:5052 request=/eth/v1/validator/duties/attester/172299 http_method=POST
rocketpool_validator  | DBG 2023-04-28 08:29:12.200+00:00 Opened connection to remote server         remote=eth2:5052 request=/eth/v1/validator/duties/attester/172299 http_method=POST connection=1
rocketpool_validator  | DBG 2023-04-28 08:29:12.201+00:00 Got REST response headers from remote server status=200 http_method=POST remote=eth2:5052 request=/eth/v1/validator/register_validator connection=4
rocketpool_validator  | DBG 2023-04-28 08:29:12.202+00:00 Beacon proposers prepared                  validators_count=46 proposers_count=45 prepared_count=1 service=duties_service
rocketpool_validator  | DBG 2023-04-28 08:29:12.204+00:00 Received REST response body from remote server remote=eth2:5052 request=/eth/v1/validator/duties/sync/172298 contentType=application/json size=40 connection=6
rocketpool_validator  | DBG 2023-04-28 08:29:12.205+00:00 REST request body has been sent            remote=eth2:5052 request=/eth/v1/validator/duties/attester/172299 size=406 http_method=POST connection=1
rocketpool_validator  | DBG 2023-04-28 08:29:12.207+00:00 No new sync committee member's duties received slot=5513546 service=duties_service
rocketpool_validator  | DBG 2023-04-28 08:29:12.207+00:00 Received REST response body from remote server remote=eth2:5052 request=/eth/v1/validator/register_validator contentType=text/plain size=0 connection=4
rocketpool_validator  | DBG 2023-04-28 08:29:12.208+00:00 Validators registered                      slot=5513546 beacon_nodes_count=1 registrations=45 validators_count=46 service=duties_service
rocketpool_validator  | DBG 2023-04-28 08:29:12.227+00:00 Got REST response headers from remote server status=200 http_method=POST remote=eth2:5052 request=/eth/v1/validator/duties/attester/172299 connection=1
rocketpool_validator  | DBG 2023-04-28 08:29:12.229+00:00 Received REST response body from remote server remote=eth2:5052 request=/eth/v1/validator/duties/attester/172299 contentType=application/json size=4338 connection=1
rocketpool_validator  | DBG 2023-04-28 08:29:12.230+00:00 No new attester's duties received          slot=5513546 service=duties_service
rocketpool_validator  | DBG 2023-04-28 08:29:13.653+00:00 Checking beacon node                       node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD status=offline service=fallback_service
rocketpool_validator  | DBG 2023-04-28 08:29:13.653+00:00 Checking beacon node status                service=fallback_service node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:13.653+00:00 Sending REST request to remote server      remote=192.168.1.187:5052 request=/eth/v1/node/version http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:13.657+00:00 Got REST response headers from remote server status=200 http_method=GET remote=192.168.1.187:5052 request=/eth/v1/node/version connection=14
rocketpool_validator  | DBG 2023-04-28 08:29:13.659+00:00 Received REST response body from remote server remote=192.168.1.187:5052 request=/eth/v1/node/version contentType=application/json size=49 connection=14
rocketpool_validator  | NOT 2023-04-28 08:29:13.660+00:00 Beacon node is online                      agent_version="prysm/v4.0.3 (linux arm64)" node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:13.661+00:00 Requesting beacon node network configuration service=fallback_service node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:13.663+00:00 Sending REST request to remote server      remote=192.168.1.187:5052 request=/eth/v1/config/spec http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:13.667+00:00 Got REST response headers from remote server status=200 http_method=GET remote=192.168.1.187:5052 request=/eth/v1/config/spec connection=14
rocketpool_validator  | DBG 2023-04-28 08:29:13.669+00:00 Received REST response body from remote server remote=192.168.1.187:5052 request=/eth/v1/config/spec contentType=application/json size=3757 connection=14
rocketpool_validator  | DBG 2023-04-28 08:29:13.672+00:00 Requesting beacon node genesis information service=fallback_service node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:13.673+00:00 Sending REST request to remote server      remote=192.168.1.187:5052 request=/eth/v1/beacon/genesis http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:13.677+00:00 Got REST response headers from remote server status=200 http_method=GET remote=192.168.1.187:5052 request=/eth/v1/beacon/genesis connection=14
rocketpool_validator  | DBG 2023-04-28 08:29:13.678+00:00 Received REST response body from remote server remote=192.168.1.187:5052 request=/eth/v1/beacon/genesis contentType=application/json size=169 connection=14
rocketpool_validator  | NOT 2023-04-28 08:29:13.678+00:00 Beacon node is compatible                  node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:13.678+00:00 Requesting beacon node sync status         service=fallback_service node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:13.678+00:00 Sending REST request to remote server      remote=192.168.1.187:5052 request=/eth/v1/node/syncing http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:13.683+00:00 Got REST response headers from remote server status=200 http_method=GET remote=192.168.1.187:5052 request=/eth/v1/node/syncing connection=14
rocketpool_validator  | DBG 2023-04-28 08:29:13.683+00:00 Received REST response body from remote server remote=192.168.1.187:5052 request=/eth/v1/node/syncing contentType=application/json size=112 connection=14
rocketpool_validator  | NOT 2023-04-28 08:29:13.684+00:00 Beacon node is in sync                     head_slot=5513546 sync_distance=0 is_optimistic=false node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
rocketpool_validator  | DBG 2023-04-28 08:29:16.000+00:00 Producing sync committee messages          delay=798us946ns slot=5513546 duties_count=0 service=sync_committee_service
rocketpool_validator  | DBG 2023-04-28 08:29:16.001+00:00 Sending REST request to remote server      remote=eth2:5052 request=/eth/v1/beacon/blocks/head/root http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:16.001+00:00 Sending REST request to remote server      remote=192.168.1.187:5052 request=/eth/v1/beacon/blocks/head/root http_method=GET
rocketpool_validator  | DBG 2023-04-28 08:29:16.001+00:00 REST client communication error            remote=192.168.1.187:5052 request=/eth/v1/beacon/blocks/head/root connection=14 errorName=HttpWriteError errorMsg="Could not send request headers"
rocketpool_validator  | DBG 2023-04-28 08:29:16.002+00:00 Unable to retrieve head block's root       node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD error="[RestCommunicationError] Communication failed while sending/receiving request, http error [HttpWriteError]: Could not send request headers" strategy=first request=getHeadBlockRoot
rocketpool_validator  | WRN 2023-04-28 08:29:16.002+00:00 Beacon node down                           node="http://192.168.1.187:5052[prysm/v4.0.3 (linux arm64)]" node_index=1 node_roles=AGBSD
```

I haven't tried the other clients yet, I can confirm it works fine again if I switch the remote BN from Prysm to Nimbus v23.4.0 though.

**To Reproduce**
1. Platform details (OS, architecture): Debian 11, `arm64`, running via Docker (the Rocket Pool stack)
2. Branch/commit used: v23.4.0
3. Commands being executed: `/home/user/nimbus_validator_client --non-interactive --beacon-node=http://eth2:5052 --beacon-node=http://192.168.1.7:5052 --data-dir=/ethclient/nimbus_vc --validators-dir=/validators/nimbus/validators --secrets-dir=/validators/nimbus/secrets --doppelganger-detection=false --suggested-fee-recipient=0xd4E96eF8eee8678dBFf4d535E033Ed1a4F7605b7 --payload-builder --metrics --metrics-address=0.0.0.0 --metrics-port=9101 --graffiti=RP-GN v1.9.3`
4. Relevant log lines: see above

**Comment by jclapis** [comment]
Update: this is still present in v23.5.0.

**Comment by alexmbird** [comment]
May be related to #4928 - please close mine if you think so.

**Comment by zah** [comment]
The issue was resolved in https://github.com/status-im/nimbus-eth2/pull/4950.
The fix was released in Nimbus 23.5.1.
